sys/net/network_layer/ng_ipv6/netif/ng_ipv6_netif.c

```diff
@@ -34,12 +34,12 @@ static ng_ipv6_netif_t ipv6_ifs[NG_NETIF_NUMOF];
 static char addr_str[NG_IPV6_ADDR_MAX_STR_LEN];
 #endif
 
-static int _add_addr_to_entry(ng_ipv6_netif_t *entry, const ng_ipv6_addr_t *addr,
-                              uint8_t prefix_len, bool anycast)
+static ng_ipv6_addr_t *_add_addr_to_entry(ng_ipv6_netif_t *entry, const ng_ipv6_addr_t *addr,
+                                          uint8_t prefix_len, uint8_t flags)
 {
     for (int i = 0; i < NG_IPV6_NETIF_ADDR_NUMOF; i++) {
         if (ng_ipv6_addr_equal(&(entry->addrs[i].addr), addr)) {
-            return 0;
+            return &(entry->addrs[i].addr);
         }
 
         if (ng_ipv6_addr_is_unspecified(&(entry->addrs[i].addr))) {
```

Comment:
Couldn't it happen that the entry already exists after the first empty entry if the list fragmented by earlier removals?